File: ClientApp/src/app/features/ships/base/classes/ship-read-resource.ts
import { KeyValuePair } from '../../../../shared/classes/keyValuePair'
import { ShipOwnerResource } from './ship-owner-resource'

export class ShipReadResource extends KeyValuePair {

    description: string
    imo: string
    flag: string
    registryNo: string
    manager: string
    managerInGreece: string
    agent: string
    isActive: boolean

    shipOwner: ShipOwnerResource

}

File: Features/Ships/Base/Resources/ShipReadResource.cs
namespace ShipCruises {

    public class ShipReadResource {

        public int Id { get; set; }
        public string Description { get; set; }
        public string IMO { get; set; }
        public string Flag { get; set; }
        public string RegistryNo { get; set; }
        public string Manager { get; set; }
        public string ManagerInGreece { get; set; }
        public string Agent { get; set; }
        public bool IsActive { get; set; }
        public string UserId { get; set; }

        public ShipOwnerReadResource ShipOwner { get; set; }

    }

}